File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) 
    {
        int min_value = Integer.MAX_VALUE;
        int low = 0;
        int high = nums.length-1;
        while(low <= high){
            int mid = (low+high)/2;
            if(nums[low] <= nums[mid]){
                min_value = Math.min(min_value, nums[low]);
                low = mid+1;
            }
            else{
                min_value = Math.min(min_value, nums[mid]);
                high = mid-1;
            }
        }
        return min_value;
        
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        ans = [float('inf')]*n
        ans[0] = 0
        for i in range(n):
            for j in range(1,nums[i]+1):
                if i+j < n:
                    ans[i+j] = min(ans[i+j],ans[i]+1)
        # print(ans)
        return int(ans[-1])

        